registry/service.go

```diff
@@ -115,6 +148,9 @@ func splitReposSearchTerm(reposName string) (string, string) {
 // Search queries the public registry for images matching the specified
 // search terms, and returns the results.
 func (s *DefaultService) Search(ctx context.Context, term string, limit int, authConfig *types.AuthConfig, userAgent string, headers map[string][]string) (*registrytypes.SearchResults, error) {
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
I think it is my mistake to lock the entire func Search. It will block others.
Now I try to only lock s.config in `Search`. Like this:
```
        s.mu.Lock()
	index, err := newIndexInfo(s.config, indexName)
	s.mu.Unlock()
```
PTAL, thanks @aaronlehmann